python_modules/dagster-test/dagster_test/toys/freshness_checks.py

```diff
@@ -0,0 +1,106 @@
+import math
+import random
+from datetime import timedelta
+
+import pendulum
+from dagster import (
+    AssetExecutionContext,
+    MetadataValue,
+    ObserveResult,
+    ScheduleDefinition,
+    TimestampMetadataValue,
+    _check as check,
+    asset,
+    build_last_update_freshness_checks,
+    build_sensor_for_freshness_checks,
+    define_asset_job,
+    observable_source_asset,
+)
+
+
+@observable_source_asset(group_name="freshness_checks")
+def unreliable_source_events():
+    """An asset which has a .5 probability of being updated every minute.
+
+    We expect an update every 3 minutes, and the asset could have arrived within the last 2 minutes.
+    """
+    return ObserveResult(
+        metadata={"dagster/last_updated_timestamp": get_last_updated_timestamp_unreliable_source()}
+    )
+
+
+@asset(non_argument_deps={unreliable_source_events.key}, group_name="freshness_checks")
+def derived_asset(context: AssetExecutionContext):
```

Comment:
better to `-> None` and not return anything, so the IO manager doesn't get needlessly invoked